File: frontend/search-ui-bun/src/app/tempo-real.models.ts
export type TempoRealTransporte = 'signalr' | 'websocket' | 'sse' | 'long-polling';

export interface TempoRealCriacaoResponse {
  tarefaId: string;
  descricao: string;
  status: string;
  versao: number;
}

export interface TempoRealEstadoResponse {
  tarefaId: string;
  descricao: string;
  status: string;
  etapa: string;
  progresso: number;
  versao: number;
  atualizadoEmUtc: string;
  mensagem: string;
}

File: src/BlogSamples/Messaging/TempoReal/TempoRealModels.cs
namespace BlogSamples.Messaging.TempoReal;

public enum TempoRealStatus
{
    Criada,
    Processando,
    Concluida,
    Falhou
}

public sealed record TempoRealCriacaoRequest(string Descricao);

public sealed record TempoRealCriacaoResponse(
    Guid TarefaId,
    string Descricao,
    string Status,
    int Versao);

public sealed record TempoRealEstadoResponse(
    Guid TarefaId,
    string Descricao,
    string Status,
    string Etapa,
    int Progresso,
    int Versao,
    DateTimeOffset AtualizadoEmUtc,
    string Mensagem);